statistics/views.py

```diff
@@ -153,10 +153,18 @@ def training(request):
     malecount = cache.get(male_key)
     if status != 0:
         if not femalecount or not malecount:
-            female_list=list(Student.objects.filter(trainingattend__training_id__in=[col.id for col in collection.qs], gender='Female').values_list('id'))
-            femalecount= len([i[0] for i in female_list])
-            male_list=list(Student.objects.filter(trainingattend__training_id__in=[col.id for col in collection.qs], gender='Male').values_list('id'))
-            malecount= len([i[0] for i in male_list])
+            training_ids = [col.id for col in collection.qs]
+            # start_time = time.time()
+            femalecount = TrainingAttend.objects.filter(training_id__in=training_ids, student__gender='Female').count()
+            malecount = TrainingAttend.objects.filter(training_id__in=training_ids, student__gender='Male').count()
```

Comment:
```suggestion
from django.db.models import Count

                        gender_counts = TrainingAttend.objects.filter(training_id__in=training_ids).values('gender').annotate(gender_count=Count('gender'))
```

You should get a dict of gender with counts